### ISSUE TITLE
Too many disconnectedCallback calls in WPT custom-elements/enqueue-custom-element-callback-reactions-inside-another-callback.html

### ISSUE THREAD

**Issue by pshaughn** [servo/servo#25014]
WPT custom-elements/enqueue-custom-element-callback-reactions-inside-another-callback.html puts a custom element inside another custom element, then removes the parent one from the DOM. Inside the parent's disconnectedCallback, the child is appended to the document body. The test is seeing the right order of operations, but it's seeing two of the child's disconnectedCallback in a row instead of one.

**Comment by pshaughn** [comment]
I'd like another pair of eyes on this; what I'm seeing seems like a spec discrepancy. 

https://dom.spec.whatwg.org/#concept-node-remove always calls disconnectedCallback on a child when a parent is removed, and always enqueues disconnectedCallback on the node itself. 
https://dom.spec.whatwg.org/#concept-node-adopt always removes the node.
Therefore, if we remove a parent and then adopt the child, we should see two disconnectedCallbacks by these steps.

However, https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element-reactions says the callback is for when the node "becomes disconnected", and that concept is defined with the phrase "and it is now no longer connected". It seems like removing an already-disconnected node possibly shouldn't get a disconnectedCallback, but I don't see anything in the specified algorithms that results in that outcome.

**Comment by jdm** [comment]
I agree, that looks to me like the two bits don't agree on when disconectedCallback is expected to be used.

**Comment by pshaughn** [comment]
https://github.com/whatwg/dom/issues/773

**Comment by jdm** [comment]
https://github.com/whatwg/dom/pull/827 updated the spec, so we should update our implementation.
